src/client/mod.rs

```diff
@@ -288,23 +309,44 @@ pub struct WrapData {
 pub struct AppRoleProperties {
     /// Require `secret_id` to be presented when logging in using this `AppRole`. Defaults to 'true'.
     pub bind_secret_id: bool,
+    /// Require `secret_id` to be presented when logging in using this `AppRole`. Defaults to 'true'.
+    pub local_secret_ids: bool,
     /// Comma-separated list of CIDR blocks; if set, specifies blocks of IP addresses which can
     /// perform the login operation.
-    pub bound_cidr_list: String,
-    /// If set, the token generated using this `AppRole` is a periodic token; so long as it is
-    /// renewed it never expires, but the TTL set on the token at each renewal is fixed to the value
-    /// specified here. If this value is modified, the token will pick up the new value at its next
-    /// renewal.
-    pub period: VaultDuration,
-    /// List of policies set on tokens issued via this `AppRole`.
-    pub policies: Vec<String>,
+    pub secret_id_bound_cidrs: Option<Vec<String>>,
     /// Number of times any particular `SecretID` can be used to fetch a token from this `AppRole`,
     /// after which the `SecretID` will expire.
     pub secret_id_num_uses: u64,
-    /// Duration after which any `SecretID` expires.
+    /// Duration in either an integer number of seconds (3600) or an integer time unit (60m) after which any SecretID expires.
     pub secret_id_ttl: VaultDuration,
+    /// List of CIDR blocks; if set, specifies blocks of IP addresses which can authenticate successfully,
+    /// and ties the resulting token to these blocks as well.
+    pub token_bound_cidrs: Vec<String>,
+    /// If set, will encode an explicit max TTL onto the token. This is a hard cap even if token_ttl and
+    /// token_max_ttl would otherwise allow a renewal.
+    pub token_explicit_max_ttl: VaultDuration,
```

Comment:
From the docstring above, it sounds like this should be an `Option<VaultDuration>`?

Reply:
Thanks for the feedback! I'll make the change to Option to be on the safe side. The documentation is from the POST since there isn't any documentation for the GET, but it looks like a default is set when performing a read. https://www.vaultproject.io/api-docs/auth/approle/#createupdate-approle

```shell
$ vault write -f auth/approle/role/empty_role
$ curl  -H "X-Vault-Request: true" -H "X-Vault-Token: $(vault print token)" -XGET http://127.0.0.1:8200/v1/auth/approle/role/empty_role | jq
{
  "request_id": "bb3e7709-1404-16dd-ba4e-f3d3047dffaf",
  "lease_id": "",
  "renewable": false,
  "lease_duration": 0,
  "data": {
    "bind_secret_id": true,
    "local_secret_ids": false,
    "secret_id_bound_cidrs": null,
    "secret_id_num_uses": 0,
    "secret_id_ttl": 0,
    "token_bound_cidrs": [],
    "token_explicit_max_ttl": 0,
    "token_max_ttl": 0,
    "token_no_default_policy": false,
    "token_num_uses": 0,
    "token_period": 0,
    "token_policies": [],
    "token_ttl": 0,
    "token_type": "default"
  },
  "wrap_info": null,
  "warnings": null,
  "auth": null
}

```